csharp/src/Ice/Communicator.cs

```diff
@@ -1055,19 +1019,22 @@ async Task PerformDisposeAsync()
                 }
                 else
                 {
-                    var adminIdentity = new Identity("admin", GetProperty("Ice.Admin.InstanceName") ?? "");
-                    if (adminIdentity.Category.Length == 0)
-                    {
-                        adminIdentity = new Identity(adminIdentity.Name, Guid.NewGuid().ToString());
-                    }
+                    string instanceName = GetProperty("Ice.Admin.InstanceName") ?? "";
```

Comment:
```suggestion
                    string instanceName = GetProperty("Ice.Admin.InstanceName") ?? Guid.NewGuid().ToString();
```

Reply:
You're right, I forgot that "" and null are the same.